service/src/main/kotlin/xyz/lebkuchenfm/api/xsounds/XSoundsEndpoint.kt

```diff
@@ -15,6 +23,42 @@ fun Route.xSoundsRouting(xSoundsService: XSoundsService) {
             val response = XSoundsResponse(sounds.map { it.toResponse() })
             call.respond(response)
         }
+
+        post {
+            var soundName = ""
+            var tags: List<String> = emptyList()
+            var fileBytes: ByteArray? = null
+
+            val multipartData = call.receiveMultipart()
+
+            multipartData.forEachPart { part ->
+                when (part) {
+                    is PartData.FormItem -> {
+                        if (part.name == "soundName") {
+                            // TODO: `soundName` validation ((null, empty, special characters, no file extension in name))
+                            soundName = part.value
+                        } else if (part.name == "tags") {
+                            tags = part.value.split(',').map { it.trim() }
+                        }
+                    }
+
+                    is PartData.FileItem -> {
+                        // TODO: handle "no file" error
+                        fileBytes = part.provider().readRemaining().readByteArray()
+                    }
+
+                    else -> {}
+                }
+                part.dispose()
+            }
+
+            fileBytes?.let { bytes ->
+                // TODO: pass authenticated user name as "addedBy"
+                val sound = xSoundsService.addNewXSound(soundName, tags, bytes)
+                val response = Json.encodeToString(XSoundResponse.serializer(), sound.toResponse())
```

Comment:
You don't have to serialize manually. You can just pass the object to `call.respond` as said in the [docs](https://ktor.io/docs/server-responses.html#object).